skyportal/models.py

```diff
@@ -80,14 +82,26 @@ class Stream(Base):
 User.groups = relationship('Group', secondary='group_users',
                            back_populates='users')
 
+
 @property
 def token_groups(self):
     return self.created_by.groups
+
+
 Token.groups = token_groups
 
 
 class Source(Base):
     id = sa.Column(sa.String, primary_key=True)
+    is_source = sa.Column(sa.Boolean, default=False)
+    is_candidate = sa.Column(sa.Boolean, default=True)
+    saved_as_source_by = relationship("User", back_populates="saved_sources")
+    saved_as_source_by_id = sa.Column(sa.ForeignKey("users.id"), nullable=True)
+    saved_as_source_at_time = sa.Column(ArrowType, nullable=True)
+    unsaved_as_source_at_time = sa.Column(ArrowType, nullable=True)
+    unsaved_as_source_by = relationship("User")
+    unsaved_as_source_by_id = sa.Column(sa.ForeignKey("users.id"), nullable=True)
```

Comment:
@dannygoldstein is there ever a time when someone could unsave a source? I think once it's made a source it should be burned in, right?

Reply:
Yes, one should be able to unsave a source. So I think it makes sense to not have the save logic update fields of the `Source` class. Probably better to have a join table that keeps track of `(source_id, filter_id, group_id, scanner_id, save_date)` 

danny

Reply:
Rows can easily be removed from that table